Release 0.2.0

uncomputable opened the pull request:
First release of rust-simplicity. Releases will make it easier to use rust-simplicity inside other projects such as elements-miniscript.

RCasatta commented on Cargo.toml:
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "simplicity"
```
to release on crates we need a different name -> https://crates.io/crates/simplicity

uncomputable commented:
Depends on #198 and #199.

RCasatta commented on simplicity-sys/Cargo.toml:
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "simplicity-sys"
```
should we change also this in `simplicity-lang-sys` ?

uncomputable replied:
I like `simplicity-sys` because it is shorter. There will be a lot of projects with the `simplicity` infix.

We invade the namespace of the existing `simplicity` project on crates.io, but the last update was 3 years ago so it seems pretty dead.

uncomputable commented:
@apoelstra Let's merge the release unless you see remaining roadblocks

RCasatta commented:
I think some mandatory fields in the `Cargo.toml` are missing such as `description` and `documentation` https://doc.rust-lang.org/cargo/reference/publishing.html#before-publishing-a-new-crate.

it can be tried out using `cargo publish --dry-run`

apoelstra commented:
On my end there are just warnings @RCasatta. I think we're ok.

apoelstra commented:
@uncomputable could you squash these three commits? (Or at least the last two; the middle one does not compile.)

uncomputable commented:
I squashed so each commit compiles.

I added the missing Cargo fields so `cargo publish` doesn't emit any warnings. Thank you @RCasatta for providing the command for dry runs!

I added the missing license to the `sys` crate. There were missing or long license headers, which I resolved. These changes belong to a separate PR, strictly speaking, but I feel like merging quickly :)

apoelstra commented:
We have a nice symmetry now with our workspaces being `simplicity-lang`, `simplicity-sys` and `simplicity-fuzz` :)

apoelstra commented:
I'm getting a bunch of clippy warnings, which maybe showed up since the edition-2021 change, but we can fix those in a followup PR.

uncomputable commented:
Unintended naming symmetry, but I like it, too

apoelstra reviewed (approved):
ACK 9bcef5155709e34b4b22379825cc10e280139d6d

apoelstra commented:
Tagged and published.

uncomputable commented:
Awesome. Cheers to the release 🎉
